packages/crayons-core/src/components/icon/icon.tsx

```diff
@@ -11,41 +22,180 @@ import icons from './icon-assets/icons.json';
 })
 export class Icon {
   /**
-   * Identifier of the icon. The attribute’s value must be a valid svg file in the repo of icons (assets/icons).
+   * Identifier of the icon. The attribute’s value must be a valid svg Name in the Crayons-Icon set.
    */
   @Prop() name: string;
-
   /**
-   * Size of the icon, specified in number of  pixels.
-   * Default value is 12px defined using the --icon-size css variable.
+   * An alternate description to use for accessibility. If omitted, the icon will be ignored by assistive devices.
+   */
+  @Prop() label: string;
+  /**
+   * Identifier of the icon. The attribute’s value must be a valid JS Import Name of the svg in the named export from @freshworks/crayons-icon.
+   */
+  @Prop() dataSvg = '';
+  /**
+   * Identifier of the icon. The attribute’s value must be a valid path to svg file.
+   */
+  @Prop() src: string;
+  /**
+   * Size of the icon, specified in number of  pixels. This will be square coordinates of (w X h) = size X size
    */
   @Prop() size: number;
-
+  /**
+   * Root Margin in px or percentage for Intersection-Observer. This means from ref to bottom of loaded view , the item loads when it crosses above the negative y margin.
+   */
+  @Prop() xRootMargin = '50px';
+  /**
+   * Width of the icon, specified in number of  pixels.
+   */
+  @Prop() width: number;
+  /**
+   * Height of the icon, specified in number of  pixels.
+   */
+  @Prop() height: number;
   /**
    * Color in which the icon is displayed, specified as a standard CSS color or as a HEX code.
    */
   @Prop() color = '';
+  /**
+   * Enable Intersection Observer. Default is false.
+   */
+  @Prop() lazy = false;
+  /**
+   * Name of External Library to be used
+   */
+  @Prop() library = 'crayons';
+
+  @Element() el: HTMLElement;
+
+  @State() private visible = false;
 
-  @State() svgHTML = '';
+  @State() private intersectionObserver: IntersectionObserver;
+
+  @State() private svgHTML = '';
+
+  @State() private eval_xOb = false;
 
   @Watch('name')
-  private setSVGState(iconName: string) {
-    this.svgHTML = icons[iconName];
+  @Watch('src')
+  @Watch('library')
+  @Watch('dataSvg')
+  @Watch('lazy')
+  private async setSVGState(): Promise<void> {
+    const { name, visible, library, lazy, dataSvg, src } = this;
+
+    if (!lazy) this.eval_xOb = true;
+    else this.eval_xOb = visible;
+
+    if (!Build.isBrowser || !this.eval_xOb) {
+      return;
+    }
+    if (!name && dataSvg) {
+      this.svgHTML = dataSvg;
+    } else if (name) {
+      try {
+        let url = src;
+        if (!src) {
+          url = this.getUrl(name, library);
+        } else url = `${src}/${name}.svg`;
+        if (url !== undefined) {
+          const svgEl = await getSVGElement(url);
+          this.applySVGMutation(library, name, svgEl);
+          this.svgHTML = svgEl.outerHTML;
+        } else {
+          throw 'Icon-URL Not Valid.';
+        }
+      } catch (ex) {
+        console.info(
```

Comment:
You are throwing an error with a message but in the below catch block , you are not logging that error but instead just having a static error description.

Reply:
Fixed